### DISCUSSION TITLE
Restricting if a chunk run depending on the output file format

### DISCUSSION THREAD

**Discussion by siganna71** [quarto-dev/quarto-cli#6665] [Q&A] [answered]
### Description

Can I restrict whether a chunk runs depending on the output file format? I.e. if I am rendering as pdf or word?
With thanks, Anna

**Comment by mcanouil** [comment, marked as answer]
See <https://quarto.org/docs/authoring/conditional.html>.

**Reply by siganna71** [reply]
Thank you so much!

**Reply by cderv** [reply]
Just a note that the computational code chunks  inside a conditional div will still be evaluated but the whole content (source and results) will be ignored from output. 

Adding this info, just so you are aware depending on what you chunk is doing.

**Comment by siganna71** [comment]
Thank you for this, that means that I can not use the same label for a figures in two chunks, one should meant for pdf and one meant for word?

**Reply by cderv** [reply]
Yes I think label can't be duplicated because both chunk would be run and unicity of label would be checked. 

Are you using knitr or Jupyter engine with Quarto ? 

If you are using R through **knitr** you could have conditional on format inside your code chunk 
Exemple: https://bookdown.org/yihui/rmarkdown-cookbook/latex-html.html
You code could then target different output according to the output format. Depending on what you want to do, it could help.

**Reply by cscheid** [reply]
This is unfortunate, because it means that there won't be a single label usable in crossreferences, right? The problem is solvable with a Lua filter (which runs before crossrefs are resolved), but it's ugly. We should try to come up with a better solution.

**Reply by cderv** [reply]
Using our conditional divs to create one chunk per format, will indeed mean that two labels so it needs also a conditional on the referenced part to use the correct label. 

Using the knitr solution (having computation code aware of output format and able to do the right thing) means one label only. So cross reference works fine.

**Reply by siganna71** [reply]
I am using knitr engine (I.e. running from R studio).  I will check this link out. 
I would love it if you could find a good solution. I love the PDF output but my advisors want to have Word files to read.

**Reply by baptiste** [reply]
@siganna71 I've found that the following workflow was useful for me (producing both html and pdf versions of a book), to tweak the figure settings for both types of output, and also use a different extension:

````

:::{.content-hidden}

```{r, echo=FALSE, results='asis', cache=FALSE}
library(ggplot2)
library(knitr)

.ar1 <- 0.6 # 1 panel, right legend
.ar2 <- 0.45 # 2 panels, right legend
.ar3 <- 0.35 # 3 panels, right legend

if (knitr::is_html_output()) {
  
.body <- 750 / 96 # px per inch
.px2pt <- 72/96 # fontsize conversion
.base_size <- 17
.ex <- unit(.base_size*.px2pt,'points') # line height

theme_set(theme_bw(base_family = 'Gill Sans Nova', base_size = .base_size*.px2pt) +
            theme(strip.background=element_blank(),
                  legend.box.margin = margin(l = -2),
                  panel.background = element_blank(),
                  axis.ticks = element_line(colour = "grey50",linewidth = rel(0.8)),
                  panel.grid.major = element_line(colour = 'grey92', linewidth = 0.5, linetype = "32"),
                  panel.grid.minor = element_line(colour = 'grey92', linewidth = 0.35, linetype = "12"),
                  panel.border = element_rect(colour='grey80', linewidth=0.8)))

update_geom_defaults("point", aes(size = 1.0, stroke=0.3)) # default `size`   -> 1.5, `stroke` -> 0.5
update_geom_defaults("line", aes(linewidth = 0.3)) # default is 0.5, too thick 
update_geom_defaults("segment", aes(linewidth = 0.3)) # default is 0.5, too thick 
update_geom_defaults("text", aes(size = 2)) # default is 3.88, too big  GeomText$default_aes

knitr::opts_chunk$set(dev = "svg", dev.args = list(bg = "white"),
                      fig.align= "center" ,
                      fig.width = .body ,
                      fig.asp =  .ar1 # default setting for height
                      )
  macros <- readLines('_macros.tex')
  mathjax <- readLines('_mathjax.tex')
  writeLines(macros)
  writeLines(mathjax)
} 

if (knitr::is_latex_output()) {
  
.body <- 5.48 # it is what it is
.ex <- unit(10,'points') # line height

theme_set(theme_bw(base_family = 'Gill Sans Nova', base_size = 10) +
            theme(strip.background=element_blank(),
                  legend.box.margin = margin(l = -2),
                  legend.key.height = .ex,
                  axis.ticks = element_line(colour = "grey50",linewidth = rel(0.8)),
                  panel.background = element_blank(),
                  panel.grid.major = element_line(colour = 'grey92', linewidth = 0.25, linetype = "32"),
                  panel.grid.minor = element_line(colour = 'grey92', linewidth = 0.2, linetype = "12"),
                  panel.border = element_rect(colour='grey80', linewidth=0.5)))
  
update_geom_defaults("point", aes(size = 0.9, stroke=0.3)) # default `size`   -> 1.5, `stroke` -> 0.5
update_geom_defaults("line", aes(linewidth = 0.3)) # default is 0.5, too thick 
update_geom_defaults("segment", aes(linewidth = 0.3)) # default is 0.5, too thick 
update_geom_defaults("text", aes(size = 2)) # default is 3.88, too big  GeomText$default_aes

knitr::opts_chunk$set(fig.width = .body,
                      fig.asp = .ar1,# default setting for height
                      fig.align= "center" ,
                      dev = "cairo_pdf", dev.args = list(bg = "white"))
} 

options("knitr.graphics.auto_pdf" = TRUE)
```

:::

````
